### PR TITLE
Bug ap 2936/column repeated

### PR DESCRIPTION
**Pull request by mbasset**
## Description

* https://quandl.atlassian.net/browse/AP-3002
* https://quandl.atlassian.net/browse/AP-2936

## Note to the reviewer

* Fix an issue when referencing dates in a QTABLE formula that would not work as expected
* Fix an issue when referencing time-series that do not exist would result in error rather than outputting no data
* Fix an issue where system would occasionally crash when trying to log errors.

## Note to QA

* [setup.zip](https://github.com/quandl/quandl-excel-windows/files/977521/setup.zip)

## Dependency

* None

### CODE REVIEW

File: Quandl.Excel.UDF.Functions/Helpers/Tools.cs [hunk 1]
```diff
@@ -162,7 +162,15 @@ public static List<object> GetValuesFromCellRange(ExcelReference excelReference)
 
         public static string GetValueFromSingleCell(ExcelReference excelReference)
         {
-            return excelReference.GetValue().ToString();
+            // This is a very basic way of detecting date formatted cell. Checking for the existence of a y as most date formats will contain at least one 'y' for the year.
```
Comment (junos) [inline Quandl.Excel.UDF.Functions/Helpers/Tools.cs]: should be `y` case sensitive?

Comment (mbasset) [inline Quandl.Excel.UDF.Functions/Helpers/Tools.cs]: I looked into that and seemed to work for me in most cases. Do you happen to know a format where y should not be case sensitive?

**Comment by ahmedn-quandl** [comment]
Verified the two issues reported are fixed tried three versions with excel, with QSeries and QTables

File: Quandl.Excel.UDF.Functions/Helpers/Tools.cs (hunk 1 above)
Comment (mbasset) [inline Quandl.Excel.UDF.Functions/Helpers/Tools.cs]: @junos is my explanation good enough?

**Comment by mbasset** [comment]
@ahmedn-quandl were there any regressions caused?

**Pull request merged**